src/markdown.c

```diff
@@ -191,6 +192,31 @@ to_html(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]) {
   return enif_make_binary(env, &output);
 }
 
+static ERL_NIF_TERM
+to_html(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[]) {
+  ErlNifBinary input;
+
+  if (enif_inspect_binary(env, argv[0], &input) == 0) {
+    return enif_make_badarg(env);
+  }
+
+  if (input.size < 1) {
+    return argv[0];
+  }
+
+  if (input.size > ARBITRARY_DIRTY_NIF_THRESHOLD) {
+#ifdef PROFILE
+      fprintf(stderr, "[markdown.c]: Input above arbitrary threshold. Running dirty NIF.\n");
```

Comment:
Does it work? I think you have to use [enif_fprintf](https://erlang.org/doc/man/erl_nif.html#enif_fprintf).

Reply:
`enif_fprintf` only gives you `%T` to print terms, right?

Reply:
I believe that it also gives you `%T` besides what you already get with a regular `fprintf`. I'm not 100% sure, but when I was writing my custom groovebox the `fprintf` didn't work and I had to always rely on `enif_fprintf`.